runtime/convertValues.go

```diff
@@ -663,10 +763,12 @@ func importCompositeValue(
 		case sema.HashAlgorithmType:
 			// HashAlgorithmType has a dedicated constructor
 			// (e.g. it has host functions)
-			return importHashAlgorithm(fields)
+			return importHashAlgorithm(inter, fields)
 
 		case sema.SignatureAlgorithmType:
-			// continue in the normal path
+			// SignatureAlgorithmType has a dedicated constructor
+			// (e.g. it has host functions)
+			return importSignatureAlgorithm(inter, fields)
```

Comment:
I didn't quite understand why the `HashAlgorithm` was handled specially, but `SignatureAlgorithm` was not. I added the special handling for `SignatureAlgorithm`, so even if this was not a bug, the code is consistent.